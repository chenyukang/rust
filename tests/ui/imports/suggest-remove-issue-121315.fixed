//@ run-rustfix
//@ compile-flags: --edition 2021
#![deny(unused_imports)]
#![allow(dead_code)]

fn test0() {
    // Test remove FlatUnused
    
    //~^ ERROR redundant import
    let _ = u32::try_from(5i32);
}

fn test1() {
    // Test remove NestedFullUnused
    
    //~^ ERROR redundant imports

    let _ = u32::try_from(5i32);
    let _a: i32 = u32::try_into(5u32).unwrap();
}

fn test2() {
    // Test remove both redundant and unused
    
    //~^ ERROR unused or redundant imports: `AsMut`, `Into`

    let _a: u32 = (5u8).into();
}

fn test3() {
    // Test remove NestedPartialUnused
    use std::convert::{Infallible};
    //~^ ERROR unused import: `From`

    trait MyTrait {}
    impl MyTrait for fn() -> Infallible {}
}

fn main() {}
