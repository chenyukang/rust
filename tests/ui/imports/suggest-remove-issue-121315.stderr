error: redundant import: `std::convert::TryFrom`
  --> $DIR/suggest-remove-issue-121315.rs:8:9
   |
LL |     use std::convert::TryFrom;
   |         ^^^^^^^^^^^^^^^^^^^^^
  --> $SRC_DIR/std/src/prelude/mod.rs:LL:COL
   |
   = note: the item `TryFrom` is already defined here
   |
note: the lint level is defined here
  --> $DIR/suggest-remove-issue-121315.rs:3:9
   |
LL | #![deny(unused_imports)]
   |         ^^^^^^^^^^^^^^

error: redundant imports: `TryFrom`, `TryInto`
  --> $DIR/suggest-remove-issue-121315.rs:15:24
   |
LL |     use std::convert::{TryFrom, TryInto};
   |                        ^^^^^^^  ^^^^^^^
  --> $SRC_DIR/std/src/prelude/mod.rs:LL:COL
   |
   = note: the item `TryFrom` is already defined here
   |
   = note: the item `TryInto` is already defined here

error: unused or redundant imports: `AsMut`, `Into`
  --> $DIR/suggest-remove-issue-121315.rs:24:24
   |
LL |     use std::convert::{AsMut, Into};
   |                        ^^^^^  ^^^^
  --> $SRC_DIR/std/src/prelude/mod.rs:LL:COL
   |
   = note: the item `Into` is already defined here

error: unused import: `From`
  --> $DIR/suggest-remove-issue-121315.rs:32:24
   |
LL |     use std::convert::{From, Infallible};
   |                        ^^^^

error: aborting due to 4 previous errors

